app/services/create_user.rb

```diff
@@ -0,0 +1,24 @@
+class CreateUser
+  attr_accessor :user
+
+  def initialize(user:)
+    self.user = user
+  end
+
+  def call
+    result = Result.new(true)
+
+    User.transaction do
```

Comment:
I think it's an excellent decision to extract this logic into its own object and not bury it in the model. Also worth calling out that the transaction logic here is what it should be: we make changes in our DB, then attempt the same in Auth0, and if that fails we roll back our transaction. I'm mentioning this as the same pattern is not followed when deleting users and I was wondering if this is deliberate.

Reply:
Good spot. I lifted the existing logic of deleting/deactivating and moved it, I am more than happy to add a transaction here too :+1: